loopy/transform/data.py

```diff
@@ -650,14 +653,45 @@ def rename_argument(kernel, old_name, new_name, existing_ok=False):
 
     kernel = smap.map_kernel(kernel)
 
+    # }}}
+
+    # {{{ args, temporary_variables
+
+    from loopy.kernel.array import ArrayBase
+    subst_mapper = SubstitutionMapper(make_subst_func(subst_dict))
+
     new_args = []
     for arg in kernel.args:
         if arg.name == old_name:
             arg = arg.copy(name=new_name)
+        if isinstance(arg, ArrayBase) and arg.shape:
+            arg = arg.copy(shape=subst_mapper(arg.shape))
```

Comment:
Stride expressions (on axes tagged that way) should also be handled. (probably also in `map_kernel`.

Reply:
Done in 30cb9fb. Handling the map_kernel discussion at https://github.com/inducer/loopy/pull/155#discussion_r504908749.